group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2011,6 +2081,42 @@ private <T> CoordinatorResult<T, CoordinatorRecord> consumerGroupFenceMember(
         }
     }
 
+    /**
+     * Write records to replace the old member by the new member.
+     *
+     * @param records   The list of records to append to.
+     * @param group     The consumer group.
+     * @param oldMember The old member.
+     * @param newMember The new member.
+     */
+    private void replaceMember(
+        List<CoordinatorRecord> records,
+        ConsumerGroup group,
+        ConsumerGroupMember oldMember,
+        ConsumerGroupMember newMember
+    ) {
+        String groupId = group.groupId();
+
+        // Remove the member without canceling its timers in case the change is reverted. If the
+        // change is not reverted, the group validation will fail and the timer will do nothing.
+        removeMember(records, groupId, oldMember.memberId());
+
+        // Generate records.
```

Comment:
I didn't see where we generated these records before. Why do we do this now?

Reply:
We did not before. For the context, we were trying to be clever in the previous implementation so we basically ensured that all the steps after the registration were triggered in order to generated all the records for the new member. We computed a new target assignment for instance. This is not really needed if the member is only replaced.

I would say that the old way was definitely more optimal but at the cost of ensuring that all the following steps understand when a new static member replaces another one. I found the latter error prone because it is pretty subtle. In this new version, I thought that we could just do the copy here and then run the following steps if they need to be ran.

My overall goal is to make the steps independent from each others as much as possible.

Reply:
Yeah I noticed the assignment code is gone as well.  So this makes sense. 
```
// Replace the current static member.
staticMemberReplaced = true;
updatedMemberBuilder = new ConsumerGroupMember.Builder(memberId)
    .setAssignedPartitions(member.assignedPartitions());    
```